restore-sql-server-transaction-logs/README.md

```diff
@@ -0,0 +1,319 @@
+
+Note that when using this option, the function ignores any object metadata.
+
+- Using metadata. To enable this functionality, set the USE_FIXED_FILE_NAME_FORMAT environment variable to "False". The function gets information about the restore from object metadata. The following metadata tags are expected:
+        
+- [CloudSqlInstance] - This is the name of the cloud sql for sql server instance where the restore request is made. Mandatory.
+
+- [DatabaseName] - This the name of the database where the function executes the restore operation. Mandatory.
+
+- [BackupType] - This is the backup type. Can be only "FULL", "DIFF" or "TLOG". Mandatory.
+
+- [Recovery] - This is the recovery type. Can be only "True" or "False". Mandatory.
+
+
+When performing the upload to the GCS bucket, construct logic to set metadata tags. In the provided powershell script, the template to place this logic is provided through functions.
+
+Note that when using this option, the function completly ignores the backup file names.
+
+The upload script provides functions where logic can be placed to define the value of the metadata tags, when using this option.
+
+1. The service account role of the Cloud SQL Instance must have objectViewer rights on the source bucket.
+
+1. The service account of the Cloud function has the following rights to import data from a cloud storage bucket to a Cloud SQL instance and to synchronize backup files on the upload and processed storage buckets:
+
+    - cloudsql.instances.get - to query information from the Cloud SQL instance
+    - cloudsql.instances.import - to request a backup import(restore) to the Cloud SQL instance
+    - eventarc.events.receiveEvent - to be able to recevie an eventarc trigger event
+    - storage.buckets.get - to be able to get information about storage buckets
+    - storage.objects.create - to be able to create objects a google cloud storage bucket
+    - storage.objects.delete - to be able to delete (overwrite) objects on a google cloud storage bucket
+    - storage.objects.get - to be able to read objects on a google cloud storage bucket
+
+1.  The powershell upload script runs in Powershell 5.1.X. or Powershell 7.X. You need the [GoogleCloud powershell module v1.0.1.10](https://cloud.google.com/tools/powershell/docs/quickstart) installed. Install [Pester](https://pester.dev/docs/quick-start) for unit tests.
+
+1.  The powershell upload script creates and uses a file called log.json. This file is important for the correct execution of the script because it is used to store reference information. It uses the reference information to upload only files newly created after the last successfully uploaded files.
+
+

```

Comment:
Add a section about how to run the tests you implemented.

Reply:
Added two short sections for running the unit tests for both the Function and the Powershell script.

Reply:
LGTM